Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm

```diff
@@ -172,20 +172,43 @@
 
 RemoteLayerTreeDrawingAreaProxy::ProcessState& RemoteLayerTreeDrawingAreaProxy::processStateForConnection(IPC::Connection& connection)
 {
-    for (auto pair : m_remotePageProcessState) {
-        if (pair.key.process().hasConnection() && &pair.key.process().connection() == &connection)
-            return pair.value;
+    for (auto& [key, value] : m_remotePageProcessState) {
+        RefPtr webProcess = WebProcessProxy::processForIdentifier(key);
+        RELEASE_ASSERT(webProcess);
+        if (webProcess->hasConnection(connection))
```

Comment:
I think this should be if (webProcess && webProcess->hasConnection(...))
A web process other than the one we received a message from may be being torn down.